distributed/tests/test_client.py

```diff
@@ -6297,3 +6297,27 @@ def log_scheduler(dask_scheduler):
     events = await c.get_events("topic2")
     assert len(events) == 2
     assert events[1][1] == ("alice", "bob")
+
+
+@gen_cluster(client=True)
+async def test_annotations(c, s, a, b):
+    da = pytest.importorskip("dask.array")
+
+    with dask.config.set(optimization__fuse__active=False):
+        with dask.annotate(priority=15, workers=[a.address], retries=2):
+            x = da.ones(10, chunks=(5,))
+
+        x = await x.persist()
+
+        assert not b.data
+        assert all("15" in str(ts.priority) for ts in s.tasks.values())
+        assert all(ts.retries == 2 for ts in s.tasks.values())
+
+        with dask.annotate(
+            workers=["fake"], allow_other_workers=True, resources={"GPU": 1}
+        ):
+            x = da.ones(10, chunks=(5,))
+
+        x = await x.persist()
+        print([{"GPU": 1} == ts.resource_restrictions for ts in s.tasks.values()])
+        assert b.data or a.data
```

Comment:
I recommend that we pull this off to a separate test, remove the print statement (or convert it to an assertion).  I'm also confused by the last assertion `assert b.data or a.data`.  What is the intent here?

Reply:
I've split these into 5 separate case, one for each distributed kwarg taxonomy (retries, priority, workers, allow_other_workers, resources).

> I'm also confused by the last assertion assert b.data or a.data.

The point was to prove that data ended up somewhere. I've removed it because its superfluous if the persist succeeded